### PR TITLE
support template option on new application generation

### PR DESCRIPTION
**Pull request by aderyabin**
I've added support `--template=slim` option on new app generation, `erb` value by default
```bash
  lotus new bookshelf --template=slim
```

### CODE REVIEW

**Comment by joneslee85** [comment]
looks good, I like it, though I think `template` is a bit confusing, I would take it as the generator skeleton templates, what  do you think about renaming it `template_engine` instead? I know it is longer but it does express its intent clearly

File: lib/lotus/cli.rb [hunk 1]
```diff
@@ -75,6 +75,7 @@ def console
     method_option :database, aliases: ['-d', '--db'], desc: "application database (#{Lotus::Generators::DatabaseConfig::SUPPORTED_ENGINES.keys.join('/')})", default: Lotus::Generators::DatabaseConfig::DEFAULT_ENGINE
     method_option :architecture, aliases: ['-a', '--arch'], desc: 'application architecture (container/app)', default: Lotus::Commands::New::Abstract::DEFAULT_ARCHITECTURE
     method_option :application_name, desc: 'application name, only for container', default: Lotus::Commands::New::Container::DEFAULT_APPLICATION_NAME
+    method_option :template, desc: 'template engine', default: 'erb'
```
Comment (joneslee85) [inline lib/lotus/cli.rb]: can we update the description with the list of available option? I really want to make it easy for users, please see line 75 for inspiration

Comment (aderyabin) [inline lib/lotus/cli.rb]: So, let's be consistent and replace `template` to `template_engine` in `.lotusrc` too

Comment (joneslee85) [inline lib/lotus/cli.rb]: well, that one might break the compatibility, I'd vote for replacing it, if we agree to do so, it's worth to mention it in the next release notes

cc @jodosha

**Comment by jodosha** [comment]
@joneslee85 @aderyabin please have a look at https://github.com/lotus/lotus/issues/155

File: lib/lotus/cli.rb (hunk 1 above)
Comment (jodosha) [inline lib/lotus/cli.rb]: @joneslee85 If we decide to proceed with this feature, we must handle `.lotusrc` as well.

**Comment by aderyabin** [comment]
Thanks for reference.

On Thu, Oct 22, 2015 at 4:26 PM, Luca Guidi <notifications@github.com>
wrote:

> @joneslee85 <https://github.com/joneslee85> @aderyabin
> <https://github.com/aderyabin> please have a look at #155
> <https://github.com/lotus/lotus/issues/155>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lotus/lotus/pull/379#issuecomment-150222983>.
>



-- 
-------------------------
Andrey Deryabin

**Comment by aderyabin** [comment]
@jodosha @AlfonsoUceda Are you still want to create a template engine gems like `lotus-haml` ?

**Comment by aderyabin** [comment]
This PR affects `Lotusrc`. It's better to accept #371 before current.

**Comment by jodosha** [comment]
> Are you still want to create a template engine gems like lotus-haml ?

@aderyabin We actually closed #155 because we do **NOT** want to maintain `lotus-haml` et. co

**Comment by aderyabin** [comment]
I think that support template engines is not so difficult:
– add templates (index.html.xxx and application.html.xxx)
– add gem to Gemfile.

@jodosha  Am I correct?

**Comment by AlfonsoUceda** [comment]
@aderyabin officially we don't want support templates but people can create gems for supporting lotus with those templates ;)

**Comment by jodosha** [comment]
@aderyabin 

The main problem is to maintain `application.html.xxx` for all that template engines: https://github.com/lotus/view#engines

The templates created by action generator are empty, so it's just a matter to specify the right extension. We take it from `.lotusrc`. If you manually edit it `template=haml` and try to generate a new action, you'll see it works fine.

Adding the gem to `Gemfile` is easy too.

**Comment by jodosha** [comment]
I'm gonna close this, because I'm afraid it can lead to inconsistencies and/or request for support for other template engines. This is something that should be eventually handled by third party gems.

**Pull request closed without merging**